iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/pattern/PipePattern.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.pipe.pattern;
+
+import javax.validation.constraints.NotNull;
+
+import java.util.Objects;
+
+public class PipePattern {
+
+  private final String pattern;
+  private final PipePatternFormat format;
+
+  public PipePattern(String pattern, @NotNull PipePatternFormat format) {
```

Comment:
Better not use "@NotNull" mark since this may introduce extra dependency. If format is null you can assign a value to it